contracts/RouterV1.sol

```diff
@@ -14,7 +15,31 @@ import {IFactory} from "./Factory/IFactory.sol";
 /// @title RouterV1
 /// @notice Convenience contract for ampleforth geyser
 /// @dev Security contact: dev-support@ampleforth.org
-contract RouterV1 {
+contract RouterV1 is IERC721Receiver {
```

Comment:
how come we need this cast?

Reply:
Need the inheritance because RouterV1 is a contract that is responsible for minting vaults (the contract will mint the vault, then transfer ownership to the user).

For RouterV1 to mint vaults, it needs to implement `onERC721Received` of the `IERC721Receiver` abstract contract. RouterV1 is checked to be a valid ERC721 receiver when `VaultFactory::create2` is called (the exact moment the check happens is when `ERC721::_safeMint` is called, which is called by `VaultFactory::create2`)

Reply:
does the check happen to see if `RouterV1` is of type `IERC721Receiver` or is it sufficient that `RouterV1` implement some of the functions of `IERC721Receiver`? If its the latter case, what do you think about defining a `implements` relationship here rather than `is a`? https://www.typescriptlang.org/docs/handbook/2/classes.html#implements-clauses

Reply:
The contract is written in Solidity, and I don't think `implements` exists. I believe that the standard method for deriving contracts is via `is` inheritance

Reply:
Alrighty! Sounds good!